src/main/kotlin/com/careem/gradle/dependencies/printer.kt

```diff
@@ -0,0 +1,98 @@
+package com.careem.gradle.dependencies
+enum class OutputType{
+    PLAIN, JSON
+}
+
+fun String.toOutputType(): OutputType = when(this){
+    "json" -> OutputType.JSON
+    else -> OutputType.PLAIN
+}
+
+
+fun print(versionDifferences: VersionDifferences, collapse: List<String>, outputType: OutputType): String {
+    return when(outputType){
+        OutputType.PLAIN -> writePlain(versionDifferences, collapse)
+        OutputType.JSON -> writeJson(versionDifferences, collapse)
+    }
+}
+
+private fun writePlain(versionDifferences: VersionDifferences, collapse: List<String>): String = buildString {
+    val (added, removed, upgraded) = versionDifferences
+    writeList("New Dependencies", added)
+    writeList("Removed Dependencies", removed, added.isNotEmpty())
+    writeList("Upgraded Dependencies", collapseDependencies(upgraded, collapse),
+        removed.isNotEmpty() || added.isNotEmpty())
+}
+
+private fun writeJson(versionDifferences: VersionDifferences, collapse: List<String>): String = buildString {
+    val (added, removed, upgraded) = versionDifferences
+    append("{")
+    writeJsonObject("new_dependencies", added, trailingComma = true)
+    writeJsonObject("removed_dependencies", removed,  trailingComma = true)
+    writeJsonObject("upgraded_dependencies", collapseDependencies(upgraded, collapse))
+    append("}")
+}
```

Comment:
this is the new function

Reply:
I didn't add any third party since this json is quite simple and adding a third party dependency doesn't add much value for this use case